atomica/model.py

```diff
@@ -1954,13 +2016,16 @@ def report_characteristic(charac, n_indent=0):
 
         # Otherwise, insert the values
         for i, c in enumerate(comps):
-            c[0] = max(0.0, x[i])
+            if self.stochastic:
+                c[0] = stochastic_rounding(max(0.0, x[i]), rng_sampler = self.rng_sampler)#integer values (still represented as floats)
```

Comment:
If the rounding is done here after checking initial condition validity is it possible that this could introduce invalid initial conditions? Or otherwise introduce discrepancies with the initial characteristic values?